src/app/widget-builder/services/widget-builder.service.ts

```diff
@@ -212,9 +213,15 @@ export class WidgetBuilderService {
   public renderWidget(widgetId: string) {
     const _self = this;
     this.lockWidgetPreview(widgetId);
-
+    const cachedLanguage = this.cache.get('currentLanguage', [widgetId], false);
+    const currentLanguage = (this.widgetPage.language) ? this.widgetPage.language : 'nl';
+    let resetCache = false;
+    if (currentLanguage !== cachedLanguage) {
+      resetCache = true;
+      this.cache.put('currentLanguage', [widgetId], currentLanguage);
+    }
```

Comment:
Nit / tip: You could simplify this a little bit like this:
```
const resetCache = currentLanguage !== cachedLanguage;
if (resetCache) {
  this.cache.put('currentLanguage', [widgetId], currentLanguage);
}
```
Doesn't make that much of a difference except that `resetCache` can be a `const` this way.

Reply:
👍 good idea